Scripts/main.js

```diff
@@ -0,0 +1,181 @@
+        var message = ruleName + '(' + ruleValue + ')' + " must be greater than " + parametr;
+        var validationResult = {
+            errorMessage: isValid ? null : message,
+            isValid: isValid
+        }
+        return validationResult;
+    });
+    return this;
+}
+
+Rule.prototype.max = function (value) {
+    var parametr = value;
+    this.functionArray.push(function (ruleValue, ruleName) {
+        var isValid = ruleValue <= parametr ? true : false;
+        var message = ruleName + '(' + ruleValue + ')' + " must be less than " + parametr;
+        var validationResult = {
+            errorMessage: isValid ? null : message,
+            isValid: isValid
+        }
+        return validationResult;
+    });
+    return this;
+}
+
+Rule.prototype.maxLength = function (value) {
+    var parametr = value;
+    this.functionArray.push(function (ruleValue, ruleName) {
+        var isValid = ruleValue.length <= parametr ? true : false;
+        var message = ruleName + "'s(" + ruleValue + ')' + " length" + '('
+            + ruleValue.length + ')' + " must be less than " + parametr;
+        var validationResult = {
+            errorMessage: isValid ? null : message,
+            isValid: isValid
+        }
+        return validationResult;
+    });
+    return this;
+}
+
+Rule.prototype.minLength = function (value) {
+    var parametr = value;
+    this.functionArray.push(function (ruleValue, ruleName) {
+        var isValid = ruleValue.length >= parametr ? true : false;
+        var message = ruleName + "'s(" + ruleValue + ')' + " length" + '('
+            + ruleValue.length + ')' + " must be greater than " + parametr;
+        var validationResult = {
+            errorMessage: isValid ? null : message,
+            isValid: isValid
+        }
+        return validationResult;
+    });
+    return this;
+}
+
+Rule.prototype.isInt = function () {
+    this.functionArray.push(function (ruleValue, ruleName) {
+        var isValid = !isNaN(parseFloat(ruleValue) && isFinite(ruleValue));
+        var message = ruleName + '(' + ruleValue + ')' + " must be a number";
+        var validationResult = {
+            errorMessage: isValid ? null : message,
+            isValid: isValid
+        }
+        return validationResult;
+    });
+    return this;
+}
+
+Rule.prototype.isEmail = function () {
+    this.functionArray.push(function (ruleValue, ruleName) {
+        var emailReg = /^[a-z][a-zA-Z0-9_.]*(\.[a-zA-Z][a-zA-Z0-9_.]*)?@[a-z][a-zA-Z0-9]*\.[a-z]+(\.[a-z]+)?$/;
+        var isValid = emailReg.test(ruleValue);
+        var message = ruleName + '(' + ruleValue + ')' + " must be an email";
+        var validationResult = {
+            errorMessage: isValid ? null : message,
+            isValid: isValid
+        }
+        return validationResult;
+    });
+    return this;
+}
+
+var Validator = function () {
+
+}
+
+Validator.validate = function (obj, rules) {
+    var properties = Object.getOwnPropertyNames(obj);
```

Comment:
I believe this line isn't necessary and code can be simplified.
var propertiesForValidation = Object.getOwnPropertyNames(rules);

_(inside the loop)_
var propertyName = propertiesForValidation[i];
var rule = rules[propertyName];
var value = obj[propertyName];

_(inside the loop)_
var validationResult = rule.functionArray[j] (value);


Is it possible to do something like this?